Packages/ClientRuntime/Sources/Networking/Http/CRT/SDKDefaultIO.swift

```diff
@@ -9,12 +9,22 @@ import AwsCommonRuntimeKit
 import class Foundation.ProcessInfo
 
 public final class SDKDefaultIO {
-    public static let shared = SDKDefaultIO()
+    static weak var privateShared: SDKDefaultIO? = nil
```

Comment:
1.  No need for `= nil` per the swift lint
2. Why is it weak?
3. Is this how we are suppose to do singletons? Seems like back in the day of obj-c we'd use dispatch_once ?